src/components/Tables/ReactiveTableRow/ReactiveTableRow.tsx

```diff
@@ -0,0 +1,89 @@
+import React from "react";
+import { EmptyRow } from "../EmptyRow";
+import { useMediaQuery } from "react-responsive";
+import { STYLES } from "../../../constants/StyleConstants";
+import "@mozilla-protocol/core/protocol/css/protocol.css";
+
+type ReactiveTableRowProps = {
+  /** the index of the row */
+  index: number;
+  /** each child provided to this element will be a column. **/
+  children?: JSX.Element[];
+  /** the column names.  on small screens, the table headers will disappear and column names will be included in the rows */
+  columnNames?: string[];
+  /** column distribution, should sum to 100%.  0 width columns or columns with no assigned width will not appear */
+  columnDistribution: string[];
+  /**  */
+  onClick?: React.MouseEventHandler;
+};
+
+const ReactiveTableRow = ({
+  index,
+  children,
+  columnNames,
+  columnDistribution,
+  onClick,
+}: ReactiveTableRowProps) => {
+  const backgroundColor = index % 2 === 0 ? STYLES.COLORS.EVEN_CELL : STYLES.COLORS.ODD_CELL;
+
+  if (!children || children.length === 0) {
+    return <EmptyRow index={index} />;
+  }
+  const isMobile = useMediaQuery({ query: `(max-width: 760px)` });
+  if (isMobile) {
+    return (
+      <div
+        style={{
+          display: "flex",
+          flex: 1,
+          flexDirection: "column",
+          padding: 16,
+          borderBottom: "1px solid #000000",
+          backgroundColor,
+        }}
+      >
+        {children?.map((childElement, index) => {
+          const columnName = columnNames ? columnNames[index] : "";
+          return (
+            <div
+              style={{
+                display: "flex",
+                justifyContent: "space-between",
+                flex: 1,
+                alignItems: "flex-start",
+              }}
+            >
+              <p
```

Comment:
I did have a question I wanted to bring up: does anyone know something similar to `htmlFor` in React?  Basically I want to link the column name and content for accessibility purposes, but I am not sure how.